ibis/backends/dask/execution/generic.py

```diff
@@ -0,0 +1,935 @@
+        how = 'first'
+
+    if how not in {'first', 'last'}:
+        raise com.OperationNotDefinedError(
+            'Arbitrary {!r} is not supported'.format(how)
+        )
+    return aggcontext.agg(data, how)
+
+
+def _filtered_reduction(mask, method, data):
+    return method(data[mask[data.index]])
+
+
+@execute_node.register(ops.Reduction, SeriesGroupBy, SeriesGroupBy)
+def execute_reduction_series_gb_mask(
+    op, data, mask, aggcontext=None, **kwargs
+):
+    method = operator.methodcaller(type(op).__name__.lower())
+    return aggcontext.agg(
+        data, functools.partial(_filtered_reduction, mask.obj, method)
+    )
+
+
+@execute_node.register(
+    (ops.CountDistinct, ops.HLLCardinality), SeriesGroupBy, SeriesGroupBy
+)
+def execute_count_distinct_series_groupby_mask(
+    op, data, mask, aggcontext=None, **kwargs
+):
+    return aggcontext.agg(
+        data,
+        functools.partial(_filtered_reduction, mask.obj, dd.Series.nunique),
+    )
+
+
+@execute_node.register(ops.Variance, SeriesGroupBy, SeriesGroupBy)
+def execute_var_series_groupby_mask(op, data, mask, aggcontext=None, **kwargs):
+    return aggcontext.agg(
+        data,
+        lambda x, mask=mask.obj, ddof=variance_ddof[op.how]: (
+            x[mask[x.index]].var(ddof=ddof)
+        ),
+    )
+
+
+@execute_node.register(ops.StandardDev, SeriesGroupBy, SeriesGroupBy)
+def execute_std_series_groupby_mask(op, data, mask, aggcontext=None, **kwargs):
+    return aggcontext.agg(
+        data,
+        lambda x, mask=mask.obj, ddof=variance_ddof[op.how]: (
+            x[mask[x.index]].std(ddof=ddof)
+        ),
+    )
+
+
+@execute_node.register(ops.Count, DataFrameGroupBy, type(None))
+def execute_count_frame_groupby(op, data, _, **kwargs):
+    result = data.size()
+    # FIXME(phillipc): We should not hard code this column name
+    result.name = 'count'
+    return result
+
+
+@execute_node.register(ops.Reduction, dd.Series, (dd.Series, type(None)))
+def execute_reduction_series_mask(op, data, mask, aggcontext=None, **kwargs):
+    operand = data[mask] if mask is not None else data
+    return aggcontext.agg(operand, type(op).__name__.lower())
+
+
+@execute_node.register(
+    (ops.CountDistinct, ops.HLLCardinality), dd.Series, (dd.Series, type(None))
+)
+def execute_count_distinct_series_mask(
+    op, data, mask, aggcontext=None, **kwargs
+):
+    return aggcontext.agg(data[mask] if mask is not None else data, 'nunique')
+
+
+@execute_node.register(ops.Arbitrary, dd.Series, (dd.Series, type(None)))
+def execute_arbitrary_series_mask(op, data, mask, aggcontext=None, **kwargs):
+    if op.how == 'first':
+        index = 0
+    elif op.how == 'last':
+        index = -1
+    else:
+        raise com.OperationNotDefinedError(
+            'Arbitrary {!r} is not supported'.format(op.how)
+        )
+
+    data = data[mask] if mask is not None else data
+    return data.iloc[index]
+
+
+@execute_node.register(ops.StandardDev, dd.Series, (dd.Series, type(None)))
```

Comment:
defnitely group thse with the others (also ok with splitting these to sub-modules), e.g. reductions / numeric